core/src/Kokkos_View.hpp

```diff
@@ -1402,22 +1402,13 @@ class View : public ViewTraits<DataType, Properties...> {
                        typename traits::array_layout> const& arg_layout,
       check_input_args check_args = check_input_args::no)
       : m_track(), m_map() {
-    // Append layout and spaces if not input
-    using alloc_prop_input = Impl::ViewCtorProp<P...>;
-
-    // use 'std::integral_constant<unsigned,I>' for non-types
-    // to avoid duplicate class error.
-    using alloc_prop = Impl::ViewCtorProp<
-        P...,
-        std::conditional_t<alloc_prop_input::has_label,
-                           std::integral_constant<unsigned int, 0>,
-                           std::string>,
-        std::conditional_t<alloc_prop_input::has_memory_space,
-                           std::integral_constant<unsigned int, 1>,
-                           typename traits::device_type::memory_space>,
-        std::conditional_t<alloc_prop_input::has_execution_space,
-                           std::integral_constant<unsigned int, 2>,
-                           typename traits::device_type::execution_space>>;
+    // Append layout and spaces if not input, split for MSVC
+    using alloc_prop_tmp =
+        decltype(Impl::add_properties(arg_prop, std::string{}));
+    using alloc_prop = decltype(
+        Impl::add_properties(std::declval<alloc_prop_tmp>(),
+                             typename traits::device_type::memory_space{},
```

Comment:
What does it even mean adding a memory space?

Reply:
This is used for the case of `create_mirror[_view]` to specify the memory space (type) we want a `View` to be allocated in.

Reply:
How so?  Isn't the memory space being deduced from the view traits analysis?

Reply:
I was talking about the overloads taking a memory space argument (possibly included in the `ViewCtorProp` argument or explicitly) for the returned view.